main/src/ca/uwaterloo/flix/language/phase/Monomorpher.scala

```diff
@@ -560,6 +560,54 @@ object Monomorpher {
           }
       }.get // This is safe since the last case can always match.
 
+    case LoweredAst.Expr.JvmReflection(exp, true, tpe, eff, loc) =>
+      val e = specializeExp(exp, env0, subst)
```

Comment:
I know you are anti-comment, but a high-level comment of what's happening in each of these cases would be nice.